src/utils/navigationConfig.tsx

```diff
@@ -1,225 +1,79 @@
-        label: "About Client-side Encryption",
-        footerNavigation: true,
-        done: false,
-      },
-      "set-up-service": {
-        key: 1,
-        label: "Verify AI service",
-        footerNavigation: true,
-        done: false,
-      },
-      "get-summary": {
-        key: 2,
-        label: "Get summary",
-        footerNavigation: true,
-        done: false,
-      },
-    },
-  },
+const generateNavigationConfig = (routePathsConfig: routePaths): NavigationConfig => {
+  /**
+   * Produces the legacy navigationConfig object from the routePathsConfig centralised object.
+   * routePathsConfig is defined in src/utils/routePathsConfig.tsx
+   */
+  const menuWithCategories: NavigationConfig = {};
+  let topIndex = 0; //
+  for (const [keyString, valueArr] of Object.entries(routePathsConfig)) {
+    // the label of the top level pages will be automatically generated from the path to avoid more complexity.
+    // If you want to avoid this behavior, add a special case in the function getLabelFromPageName in src/utils/navigationConfig.tsx
+    menuWithCategories[keyString] = {
+      key: topIndex++,
+      label: routePathsConfig[keyString][0].label || getLabelFromPageName(keyString),
+    };
+    if (valueArr.length > 1) {
+      for (const [btmIndex, page] of valueArr.entries()) {
+        menuWithCategories[keyString] = {
+          ...menuWithCategories[keyString],
+          [page.path]: {
+            key: btmIndex,
+            label: page.label,
+            footerNavigation: true,
+            done: false,
+          },
+        };
+      }
+    }
+  }
+  return menuWithCategories;
 };
+
+export const generateComponentsList = (routePathsConfig: routePaths): Record<string, JSX.Element> => {
+  /**
+   * Produces the componentsList object from the routePathsConfig centralised object.
+   * routePathsConfig is defined in src/utils/routePathsConfig.tsx
+   * The componentsList object is used in src/Router.tsx to generate the navigation routes.
+   * It simply maps the path to the component to render.
+   */
+  const componentsList: Record<string, JSX.Element> = {};
+  for (const [key, value] of Object.entries(routePathsConfig)) {
+    if (routePathsConfig[key].length === 1) {
+      componentsList[key] = value[0].component || <>No component Provided</>;
+      continue;
+    }
+    for (let i = 0; i < value.length; i++) {
+      componentsList[key + "/" + value[i].path] = value[i].component || <>No component Provided</>;
+    }
+  }
+  return componentsList;
+};
+
+function getLabelFromPageName(key: string): string {
+  /**
+   * fallback utility function if a label is not provided in the routePathsConfig
+   */
+  const specialCases: Record<string, string> = {
+    "client-side-encryption": "Encrypt Client-Side",
+  };
+  if (specialCases[key]) return specialCases[key];
+  return key
+    .replace(/-/g, " ")
+    .replace(/\b\w/g, (c) => c.toUpperCase())
+    .replace(/vm/g, "VM")
+    .replace(/cosmian/g, "Cosmian");
+}
```

Comment:
Why don't we simply make label mandatory - and remove this function ? This seems a little over-engineered and could be handle defining a type for navigation content

Reply:
the purpose of this function has been redefined and now it serves only as fallback, what do you think ? We can still delete but we should make label mandatory and hence more checks

The idea was to have the least possible "mandatory" fields in our object to make updating the codebase easier later on if someone has yet another thing to add on top of what we already have